### PR TITLE
Deployments upgrade to HttpClient

### PR DESCRIPTION
**Pull request by andrewazores**
This PR upgrades the DeploymentApiService from the old Angular Http to the newer HttpClient, and does some light refactoring on top.

### CODE REVIEW

**Comment by joshuawilson** [comment]
I think you need to update the `src/app/app.module.ts` too.

**Comment by andrewazores** [comment]
@joshuawilson you mean to add the module import for the HttpClientModule, right? Or something else?

Should that module be added to the top-level app.module.ts now, or restricted to the DeploymentsModule?

**Comment by joshuawilson** [comment]
I would think it best to add it to the top level so that we don't end up with it left behind in deployments. As long as you can have both old and new `http` apis co-exist. The whole app needs to move to that anyway.

File: src/app/space/create/deployments/services/deployment-api.service.ts [hunk 1]
```diff
@@ -134,79 +136,83 @@ export interface SeriesData {
...
-    return this.http.delete(url, { headers: this.headers })
-      .catch((err: Response) => this.handleHttpError(err));
+  deleteDeployment(spaceId: string, environmentName: string, applicationId: string): Observable<void> {
+    const encSpaceId: string = encodeURIComponent(spaceId);
+    const encEnvironmentName: string = encodeURIComponent(environmentName);
+    const encApplicationId: string = encodeURIComponent(applicationId);
+    const url: string = `${this.apiUrl}${encSpaceId}/applications/${encApplicationId}/deployments/${encEnvironmentName}`;
+    return this.http.delete(url, { headers: this.headers, responseType: 'text' })
+      .catch((err: HttpErrorResponse) => this.handleHttpError(err))
+      .map(() => null);
```
Comment (rohitkrai03) [inline src/app/space/create/deployments/services/deployment-api.service.ts]: @andrewazores Why do we need to map the value to `null` here?

Comment (rohitkrai03) [reply src/app/space/create/deployments/services/deployment-api.service.ts]: Got it.

**Pull request merged**